ecrtokenrefresher/pkg/aws/aws.go

```diff
@@ -25,20 +25,61 @@ const (
 	defaultAccountID   = "783794618700"
 	devAccountID       = "857151390494"
 	envRegionName      = "AWS_REGION"
-	regionDefault      = "us-west-2"
 )
 
-func GetECRCredentials() ([]ECRAuth, error) {
-	// Default AWS Region to us-west-2 if not set by User.
-	_, ok := os.LookupEnv(envRegionName)
-	if !ok {
-		err := os.Setenv(envRegionName, regionDefault)
+func GetECRCredentials() ([]ECRAuth, []string, error) {
+	var creds []ECRAuth
+	var err error
+	failedList := make([]string, 0)
+	for _, region := range getSupportedRegions() {
+		regionCreds, err := getECRCredentialByRegion(region)
 		if err != nil {
-			return nil, err
+			failedList = append(failedList, region)
 		}
+		creds = append(creds, regionCreds...)
+	}
+	if len(creds) < 1 {
+		return nil, failedList, fmt.Errorf("failed for all regions, %v", err)
 	}
+	return creds, failedList, nil
+}
 
```

Comment:
don't they supply the region in the aws creds?

Reply:
So this is for replication of the secret in all regions. Currently the ask is to support all regions. Since copying the token and changing the endpoint doesnt work we would need to request a new token for each region.

Reply:
Yeh, supporting all regions to me doesn't mean supporting them all at the same time to me at least. The user would specify what region they want to use.